test/test_bit_storage.cpp

```diff
@@ -0,0 +1,133 @@
+#include "../src/bitstorage/base64_bit_storage.h"
+#include "../src/bitstorage/bech32m_bit_storage.h"
+#include "../src/bech32m_exception.h"
+#include "../src/bitstorage/bin_bit_storage.h"
+#include "test_macros.h"
+
+// counter for failed errors
+extern long errors_count;
+extern long success_count;
+
+/**
+ * TODO: more tests
+ */
+void test_hex_bit_storage() {
+    // f    a    f    a    f    a
+    // 1111 1010 1111 1010 1111 1010
+    // 11111 01011 11101 01111 10100
+    const std::string to_encode = "fafafa";
+    Bech32mBitStorage storage = HexBitStorage(to_encode);
```

Comment:
Your coverity suggests that there is uncaught exeption, as those are test i sugest adding line before line 19 that would assert exeption was not thrown. I saw you have macro with this functionality but it catches only std exeption line would look like this `    ASSERT_DOES_NOT_THROW(HexBitStorage(to_encode));
` if you dont want to change macro then i sugest replacing line 19 content with block 
`    try {
        Bech32mBitStorage storage = HexBitStorage(to_encode);
    } catch (Bech32mException e) {
        ASSERT_EQUALS(true, false);
    }`